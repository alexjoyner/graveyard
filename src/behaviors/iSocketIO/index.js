import { env } from "../../.env";
const io = require('socket.io-client');
const socket = io(env.serverAddr);

class iSocketIO {
  constructor(){
    if(this.subscribe === undefined)
      throw new TypeError("Must override method");
    if(this.unsubscribe === undefined)
      throw new TypeError("Must override method");
    if(this.join === undefined)
      throw new TypeError("Must override method");
  }
}

class PointsSocket extends iSocketIO {
  constructor(){super()}
  subscribe(pointsArray, cb){
    this.join(pointsArray);
    socket.on('add log', (log) => {
        cb(null, log);
    })
  }
  unsubscribe(pointsArray, cb){
    socket.emit('leave-group', pointsArray);
  }
  join(pointsArray){
    socket.emit('join-group', pointsArray);
  }
}


export {PointsSocket};