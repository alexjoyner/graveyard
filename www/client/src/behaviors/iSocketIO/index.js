import io from 'socket.io-client';
import { env } from '../../.env';

export class ISocketIO {
  constructor() {
    if (this.subscribe === undefined) { throw new TypeError('Must override method'); }
    if (this.unsubscribe === undefined) { throw new TypeError('Must override method'); }
    if (this.join === undefined) { throw new TypeError('Must override method'); }
  }
}

class PointsSocket extends ISocketIO {
  constructor(opts) {
    super();
    this.socket = opts.socket || /* istanbul ignore next */ io(env.serverAddr);
  }
  subscribe(pointsArray, cb) {
    this.points = pointsArray;
    this.join();
    this.socket.on('add log', /* istanbul ignore next */ (log) => {
      cb(null, log);
    });
  }
  unsubscribe() {
    this.socket.emit('leave-group', this.points);
  }
  join() {
    this.socket.emit('join-group', this.points);
  }
  setPoints(pointsArray) {
    this.points = pointsArray;
  }
  getPoints() {
    return this.points;
  }
}

export { PointsSocket };
