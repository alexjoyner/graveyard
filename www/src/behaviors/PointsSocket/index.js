import io from 'socket.io-client';
import { env } from '../../.env';
import { ISocketIO } from '../interfaces/ISocketIO';

class PointsSocket extends ISocketIO {
  constructor() {
    super();
    this.points = [];
    const connectionOptions = {
      'force new connection': true,
      reconnectionAttempts: 'Infinity', // avoid having user reconnect manually in order to prevent dead clients after a server restart
      timeout: 10000, // before connect_error and connect_timeout are emitted.
      transports: ['websocket'],
    };
    this.socket = io(env.serverAddr, connectionOptions);
  }

  subscribe(pointsArray, cb) {
    this.points = pointsArray;
    this.socket.on('add log', /* istanbul ignore next */ (log) => {
      cb(null, log);
    });
    this.join();
    this.getLastLogs();
  }

  unsubscribe() {
    if (this.points.length > 0) {
      this.socket.emit('leave-group', this.points);
    }
  }

  join() {
    this.socket.emit('join-group', this.points);
  }

  getLastLogs() {
    this.socket.emit('get-last-logs', this.points);
  }

  setPoints(pointsArray) {
    this.points = pointsArray;
  }

  getPoints() {
    return this.points;
  }
}

export { PointsSocket };
