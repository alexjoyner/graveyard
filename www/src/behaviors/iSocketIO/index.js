import io from 'socket.io-client';
import { env } from '../../.env';

export class ISocketIO {
  constructor() {
    if (this.subscribe === undefined) { throw new TypeError('Must override method'); }
    if (this.unsubscribe === undefined) { throw new TypeError('Must override method'); }
    if (this.join === undefined) { throw new TypeError('Must override method'); }
  }
}

class PointsSocket extends ISocketIO {
  constructor(opts) {
    super();
    this.socket = opts.socket || /* istanbul ignore next */ io(env.serverAddr);
  }

  subscribe(pointsArray, cb) {
    this.points = pointsArray;
    this.socket.on('add log', /* istanbul ignore next */ (log) => {
      cb(null, log);
    });
    this.join();
    this.getLastLogs();
  }

  unsubscribe() {
    this.socket.emit('leave-group', this.points);
  }

  join() {
    this.socket.emit('join-group', this.points);
  }

  getLastLogs() {
    this.socket.emit('get-last-logs', this.points);
  }

  setPoints(pointsArray) {
    this.points = pointsArray;
  }

  getPoints() {
    return this.points;
  }
}

export { PointsSocket };
