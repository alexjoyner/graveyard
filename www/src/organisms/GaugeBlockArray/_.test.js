import React from 'react';
import io from 'socket.io-client';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';
import { BaseGaugeBlockArray } from '.';
import { PointsSocket } from '../../behaviors/iSocketIO';

const mockFetch = data => jest.fn().mockImplementation(() => Promise.resolve({
  status: 200,
  json: () => data,
}));

jest.mock('socket.io-client', () => ({
  on: jest.fn(),
  emit: jest.fn(),
}));

window.fetch = mockFetch([
  { id: 1, name: 'testPoint' },
  { id: 2, name: 'testPoint2' },
]);

describe('BaseGaugeBlockArray component', () => {
  describe('Component without data', () => {
    let component;
    let props;
    beforeEach(() => {
      props = {
        publishNewPoints: jest.fn(),
        publishNewLog: jest.fn(),
        removeAllPoints: jest.fn(),
        socket: new PointsSocket({
          socket: io,
        }),
        points: {},
      };
      component = shallow(<BaseGaugeBlockArray {...props} />);
    });
    it('Should render without exploding', () => {
      expect(component).toBeDefined();
    });
    it('Should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
  describe('Component with points', () => {
    let component;
    let props;
    beforeEach(() => {
      props = {
        publishNewPoints: jest.fn(),
        publishNewLog: jest.fn(),
        removeAllPoints: jest.fn(),
        socket: new PointsSocket({
          socket: io,
        }),
        points: {
          1: { name: 'Test Input', value: 10 },
        },
      };
      component = shallow(<BaseGaugeBlockArray {...props} />);
    });
    it('Should render without exploding', () => {
      expect(component).toBeDefined();
    });
    it('Should match snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
  describe('Full Mounted Component', () => {
    let props;
    let Base;
    beforeEach(() => {
      const mockStore = {
        subscribe: jest.fn(),
        dispatch: jest.fn(),
        getState: jest.fn(),
      };
      props = {
        publishNewPoints: jest.fn(),
        publishNewLog: jest.fn(),
        removeAllPoints: jest.fn(),
        socket: new PointsSocket({
          socket: io,
        }),
        currentGroup: 0,
        points: {
          1: { name: 'Test Input', value: 10 },
        },
      };
      Base = () => (
        <Provider store={mockStore}>
          <BaseGaugeBlockArray {...props} />
        </Provider>
      );
    });
    it('should call componentDidUpdate', () => {
      const spy = jest.spyOn(BaseGaugeBlockArray.prototype, 'componentDidUpdate');
      const component = mount(<Base />);
      component.setProps({ ...props });
      expect(spy).toHaveBeenCalled();
    });
    it('should call handleRemoveAllPoints', () => {
      const spy = jest.spyOn(BaseGaugeBlockArray.prototype, 'handleRemoveAllPoints');
      const root = mount(<Base />);
      const component = root.find('BaseGaugeBlockArray');
      component.instance().componentDidUpdate({
        ...props,
        currentGroup: 10,
      });
      expect(spy).toHaveBeenCalled();
    });
    it('should call handleSubscribeToGroup', () => {
      const spy = jest.spyOn(BaseGaugeBlockArray.prototype, 'handleSubscribeToGroup');
      const root = mount(<Base />);
      const component = root.find('BaseGaugeBlockArray');
      component.instance().componentDidUpdate({
        ...props,
        currentGroup: 10,
      });
      expect(spy).toHaveBeenCalled();
    });
  });
});
