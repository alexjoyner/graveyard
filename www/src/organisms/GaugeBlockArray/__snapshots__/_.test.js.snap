// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseGaugeBlockArray component Component with points Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BaseGaugeBlockArray
    points={
      Object {
        "1": Object {
          "name": "Test Input",
          "value": 10,
        },
      }
    }
    publishNewLog={[MockFunction]}
    publishNewPoints={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "1": Object {
                "id": 1,
                "name": "testPoint",
              },
              "2": Object {
                "id": 2,
                "name": "testPoint2",
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    removeAllPoints={[MockFunction]}
    socket={
      PointsSocket {
        "points": Array [
          "1",
          "2",
        ],
        "socket": Object {
          "emit": [MockFunction] {
            "calls": Array [
              Array [
                "join-group",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "get-last-logs",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "join-group",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "get-last-logs",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "join-group",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "get-last-logs",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "join-group",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "get-last-logs",
                Array [
                  "1",
                  "2",
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "on": [MockFunction] {
            "calls": Array [
              Array [
                "add log",
                [Function],
              ],
              Array [
                "add log",
                [Function],
              ],
              Array [
                "add log",
                [Function],
              ],
              Array [
                "add log",
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(BasePointGaugeBlock)
          id="1"
          point={
            Object {
              "name": "Test Input",
              "value": 10,
            }
          }
          points={
            Object {
              "1": Object {
                "name": "Test Input",
                "value": 10,
              },
            }
          }
          publishNewLog={[MockFunction]}
          publishNewPoints={
            [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "1": Object {
                      "id": 1,
                      "name": "testPoint",
                    },
                    "2": Object {
                      "id": 2,
                      "name": "testPoint2",
                    },
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          removeAllPoints={[MockFunction]}
          socket={
            PointsSocket {
              "points": Array [
                "1",
                "2",
              ],
              "socket": Object {
                "emit": [MockFunction] {
                  "calls": Array [
                    Array [
                      "join-group",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "get-last-logs",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "join-group",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "get-last-logs",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "join-group",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "get-last-logs",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "join-group",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "get-last-logs",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                },
                "on": [MockFunction] {
                  "calls": Array [
                    Array [
                      "add log",
                      [Function],
                    ],
                    Array [
                      "add log",
                      [Function],
                    ],
                    Array [
                      "add log",
                      [Function],
                    ],
                    Array [
                      "add log",
                      [Function],
                    ],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                },
              },
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "id": "1",
          "point": Object {
            "name": "Test Input",
            "value": 10,
          },
          "points": Object {
            "1": Object {
              "name": "Test Input",
              "value": 10,
            },
          },
          "publishNewLog": [MockFunction],
          "publishNewPoints": [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "1": Object {
                    "id": 1,
                    "name": "testPoint",
                  },
                  "2": Object {
                    "id": 2,
                    "name": "testPoint2",
                  },
                },
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "removeAllPoints": [MockFunction],
          "socket": PointsSocket {
            "points": Array [
              "1",
              "2",
            ],
            "socket": Object {
              "emit": [MockFunction] {
                "calls": Array [
                  Array [
                    "join-group",
                    Array [
                      "1",
                      "2",
                    ],
                  ],
                  Array [
                    "get-last-logs",
                    Array [
                      "1",
                      "2",
                    ],
                  ],
                  Array [
                    "join-group",
                    Array [
                      "1",
                      "2",
                    ],
                  ],
                  Array [
                    "get-last-logs",
                    Array [
                      "1",
                      "2",
                    ],
                  ],
                  Array [
                    "join-group",
                    Array [
                      "1",
                      "2",
                    ],
                  ],
                  Array [
                    "get-last-logs",
                    Array [
                      "1",
                      "2",
                    ],
                  ],
                  Array [
                    "join-group",
                    Array [
                      "1",
                      "2",
                    ],
                  ],
                  Array [
                    "get-last-logs",
                    Array [
                      "1",
                      "2",
                    ],
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              },
              "on": [MockFunction] {
                "calls": Array [
                  Array [
                    "add log",
                    [Function],
                  ],
                  Array [
                    "add log",
                    [Function],
                  ],
                  Array [
                    "add log",
                    [Function],
                  ],
                  Array [
                    "add log",
                    [Function],
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              },
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "CenteredContent-ys47ff-0",
        "isStatic": true,
        "rules": Array [
          "text-align:center;",
        ],
      },
      "displayName": "CenteredContent",
      "render": [Function],
      "styledComponentId": "CenteredContent-ys47ff-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(BasePointGaugeBlock)
            id="1"
            point={
              Object {
                "name": "Test Input",
                "value": 10,
              }
            }
            points={
              Object {
                "1": Object {
                  "name": "Test Input",
                  "value": 10,
                },
              }
            }
            publishNewLog={[MockFunction]}
            publishNewPoints={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "1": Object {
                        "id": 1,
                        "name": "testPoint",
                      },
                      "2": Object {
                        "id": 2,
                        "name": "testPoint2",
                      },
                    },
                  ],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
            removeAllPoints={[MockFunction]}
            socket={
              PointsSocket {
                "points": Array [
                  "1",
                  "2",
                ],
                "socket": Object {
                  "emit": [MockFunction] {
                    "calls": Array [
                      Array [
                        "join-group",
                        Array [
                          "1",
                          "2",
                        ],
                      ],
                      Array [
                        "get-last-logs",
                        Array [
                          "1",
                          "2",
                        ],
                      ],
                      Array [
                        "join-group",
                        Array [
                          "1",
                          "2",
                        ],
                      ],
                      Array [
                        "get-last-logs",
                        Array [
                          "1",
                          "2",
                        ],
                      ],
                      Array [
                        "join-group",
                        Array [
                          "1",
                          "2",
                        ],
                      ],
                      Array [
                        "get-last-logs",
                        Array [
                          "1",
                          "2",
                        ],
                      ],
                      Array [
                        "join-group",
                        Array [
                          "1",
                          "2",
                        ],
                      ],
                      Array [
                        "get-last-logs",
                        Array [
                          "1",
                          "2",
                        ],
                      ],
                    ],
                    "results": Array [
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                    ],
                  },
                  "on": [MockFunction] {
                    "calls": Array [
                      Array [
                        "add log",
                        [Function],
                      ],
                      Array [
                        "add log",
                        [Function],
                      ],
                      Array [
                        "add log",
                        [Function],
                      ],
                      Array [
                        "add log",
                        [Function],
                      ],
                    ],
                    "results": Array [
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                      Object {
                        "isThrow": false,
                        "value": undefined,
                      },
                    ],
                  },
                },
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "id": "1",
            "point": Object {
              "name": "Test Input",
              "value": 10,
            },
            "points": Object {
              "1": Object {
                "name": "Test Input",
                "value": 10,
              },
            },
            "publishNewLog": [MockFunction],
            "publishNewPoints": [MockFunction] {
              "calls": Array [
                Array [
                  Object {
                    "1": Object {
                      "id": 1,
                      "name": "testPoint",
                    },
                    "2": Object {
                      "id": 2,
                      "name": "testPoint2",
                    },
                  },
                ],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
            "removeAllPoints": [MockFunction],
            "socket": PointsSocket {
              "points": Array [
                "1",
                "2",
              ],
              "socket": Object {
                "emit": [MockFunction] {
                  "calls": Array [
                    Array [
                      "join-group",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "get-last-logs",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "join-group",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "get-last-logs",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "join-group",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "get-last-logs",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "join-group",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                    Array [
                      "get-last-logs",
                      Array [
                        "1",
                        "2",
                      ],
                    ],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                },
                "on": [MockFunction] {
                  "calls": Array [
                    Array [
                      "add log",
                      [Function],
                    ],
                    Array [
                      "add log",
                      [Function],
                    ],
                    Array [
                      "add log",
                      [Function],
                    ],
                    Array [
                      "add log",
                      [Function],
                    ],
                  ],
                  "results": Array [
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                    Object {
                      "isThrow": false,
                      "value": undefined,
                    },
                  ],
                },
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "CenteredContent-ys47ff-0",
          "isStatic": true,
          "rules": Array [
            "text-align:center;",
          ],
        },
        "displayName": "CenteredContent",
        "render": [Function],
        "styledComponentId": "CenteredContent-ys47ff-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`BaseGaugeBlockArray component Component without data Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BaseGaugeBlockArray
    points={Object {}}
    publishNewLog={[MockFunction]}
    publishNewPoints={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "1": Object {
                "id": 1,
                "name": "testPoint",
              },
              "2": Object {
                "id": 2,
                "name": "testPoint2",
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    removeAllPoints={[MockFunction]}
    socket={
      PointsSocket {
        "points": Array [
          "1",
          "2",
        ],
        "socket": Object {
          "emit": [MockFunction] {
            "calls": Array [
              Array [
                "join-group",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "get-last-logs",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "join-group",
                Array [
                  "1",
                  "2",
                ],
              ],
              Array [
                "get-last-logs",
                Array [
                  "1",
                  "2",
                ],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "on": [MockFunction] {
            "calls": Array [
              Array [
                "add log",
                [Function],
              ],
              Array [
                "add log",
                [Function],
              ],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
